setup_dependencies.sh

```diff
@@ -0,0 +1,162 @@
+#
+function error () {
+    check_result 255 "$*"
+}
+
+#
+# Usage: info <INFO_MESSAGE_1> [ <INFO_MESSAGE_2> ... ]
+#
+function info () {
+    echo "[OCTF][INFO] $*" 1>&2
+}
+
+function detect_distribution ()
+{
+    if [ -f /etc/redhat-release ] || [ -f /etc/centos-release ]
+    then
+        if ( cat /etc/redhat-release | grep "Red Hat Enterprise Linux Server release 7." &>/dev/null ) || \
+                ( cat /etc/centos-release | grep "CentOS Linux release 7." &>/dev/null )
+        then
+            echo RHEL7
+            return 0
+        fi
+    fi
+
+    if [ -f /etc/os-release ]
+    then
+        if ( cat /etc/os-release | grep "Ubuntu 18" &>/dev/null )
+        then
+            echo UBUNTU18
+            return 0
+        fi
+    fi
+
+    return 1
+}
+
+function setup_protocol_buffer
+{
+    info "Setup google protocol buffer"
+    rm -rf build/protobuf
+    mkdir -p build
+    pushd build
+
+    git clone https://github.com/protocolbuffers/protobuf.git
+    check_result $? "Cannot clone google protocol buffer gtest gtest-devel"
+    pushd protobuf
+
+    latest_release=$(curl --silent "https://github.com/protocolbuffers/protobuf/releases/latest" | sed 's#.*tag/\(.*\)\".*#\1#')
+    if [ "${latest_release}" == "" ]
+    then
+        error "Cannot get latest version of google protocol buffer"
+    fi
+
+    git checkout ${latest_release}
+    check_result $? "Cannot checkout version ${latest_release}"
+
+    git submodule update --init --recursive && \
+        ./autogen.sh && \
+        ./configure && \
+        make -j$(nproc) && \
+        make check -j$(nproc) && \
+        make install -j$(nproc) && \
+        ldconfig
+    check_result $? "Cannot setup google protocol buffer"
+
+    popd
+    popd
+
+    rm -rf build/protobuf
+}
+
+function setup_rhel7 ()
+{
+    info "Install EPEL repository"
+    yum -y install epel-release
+    check_result $? "Cannot Install EPEL repository"
+
+    info "Setup cmake 3"
+    yum -y install cmake3 && \
+    alternatives --install /usr/local/bin/cmake cmake /usr/bin/cmake 10 \
+        --slave /usr/local/bin/ctest ctest /usr/bin/ctest \
+        --slave /usr/local/bin/cpack cpack /usr/bin/cpack \
+        --slave /usr/local/bin/ccmake ccmake /usr/bin/ccmake \
+        --family cmake \
+    && \
+    alternatives --install /usr/local/bin/cmake cmake /usr/bin/cmake3 20 \
+        --slave /usr/local/bin/ctest ctest /usr/bin/ctest3 \
+        --slave /usr/local/bin/cpack cpack /usr/bin/cpack3 \
+        --slave /usr/local/bin/ccmake ccmake /usr/bin/ccmake3 \
+        --family cmake
+
+    check_result $? "Cannot setup alternative for cmake 3"
+
+    setup_protocol_buffer
+}
+
+function setup_ubuntu18 ()
+{
+    info "Setup google test"
+    rm -rf build/gtest
+    mkdir -p build/gtest
+    pushd build/gtest
+    cmake /usr/src/gtest && \
+    make -j$(nproc) && \
+    make install -j$(nproc)
+    check_result $? "Cannot setup google test"
+}
+
+DISTRO=$(detect_distribution)
```

Comment:
I suggest using lowercase naming convention to avoid conflicts with environment or shell variables.